lib/requestIterator.js

```diff
@@ -106,9 +106,19 @@ RequestIterator.prototype.rebuildRequest = function () {
   return data
 }
 
-RequestIterator.prototype.recordBody = function (request, status, body) {
+RequestIterator.prototype.recordBody = function (request, status, body, headers) {
   if (request && typeof request.onResponse === 'function') {
-    request.onResponse(status, body, this.context)
+    const toHeaderKeyValue = (headersArray) => {
+      const headersKeyValue = {}
+
+      for (let i = 0; i < headersArray.length - 1; ++i) {
+        headersKeyValue[headersArray[i]] = headersArray[i + 1]
+      }
+
+      return headersKeyValue
+    }
```

Comment:
by top level I mean to not allocate a closure and call a function defined outside of this one.